airbyte-webapp/src/components/connectorBuilder/types.ts

```diff
@@ -230,6 +252,80 @@ const nonPathRequestOptionSchema = yup
 // eslint-disable-next-line no-useless-escape
 export const timeDeltaRegex = /^(([\.\d]+?)y)?(([\.\d]+?)m)?(([\.\d]+?)w)?(([\.\d]+?)d)?$/;
 
+const regularSlicerShape = {
+  cursor_field: yup.mixed().when("type", {
+    is: (val: string) => val !== "SubstreamSlicer" && val !== "CartesianProductStreamSlicer",
+    then: yup.string().required("form.empty.error"),
+    otherwise: (schema) => schema.strip(),
+  }),
+  slice_values: yup.mixed().when("type", {
+    is: "ListStreamSlicer",
+    then: yup.array().of(yup.string()),
+    otherwise: (schema) => schema.strip(),
+  }),
+  request_option: nonPathRequestOptionSchema,
```

Comment:
I don't think this is quite correct - it seems like `path` is allowed as the `Inject into` value for a Substream slicer, but not allowed for List or Datetime. So we will need to encode that here in the yup schema

Reply:
I checked the code and while the path injection is permitted it doesn't look like it has any effect (the slicer is simply ignoring it). This is how I tested:
```
version: 0.1.0
type: DeclarativeSource
check:
  type: CheckStream
  stream_names: []
streams:
  - type: DeclarativeStream
    name: moves
    primary_key: []
    retriever:
      type: SimpleRetriever
      name: moves
      primary_key: []
      requester:
        type: HttpRequester
        name: moves
        url_base: https://pokeapi.co/api/v2/
        path: >-
          pokemon//{{
          stream_slice.the_renamed_value_to_interpolate_somewhere_else }}
        request_options_provider:
          request_parameters: {}
          request_headers: {}
          request_body_json: {}
        authenticator:
          type: NoAuth
      record_selector:
        type: RecordSelector
        extractor:
          type: DpathExtractor
          field_pointer:
            - moves
      paginator:
        type: NoPagination
      stream_slicer:
        type: SubstreamSlicer
        parent_stream_configs:
          - type: ParentStreamConfig
            parent_key: name
            request_option:
              inject_into: path
              type: RequestOption
            stream_slice_field: the_renamed_value_to_interpolate_somewhere_else
            stream:
              type: DeclarativeStream
              name: pokemon
              primary_key: []
              retriever:
                type: SimpleRetriever
                name: pokemon
                primary_key: []
                requester:
                  type: HttpRequester
                  name: pokemon
                  url_base: https://pokeapi.co/api/v2/
                  path: pokemon/
                  request_options_provider:
                    request_parameters:
                      limit: '3'
                    request_headers: {}
                    request_body_json: {}
                  authenticator:
                    type: NoAuth
                record_selector:
                  type: RecordSelector
                  extractor:
                    type: DpathExtractor
                    field_pointer:
                      - results
                paginator:
                  type: NoPagination
spec:
  connection_specification:
    $schema: http://json-schema.org/draft-07/schema#
    type: object
    required: []
    properties: {}
    additionalProperties: true
  documentation_url: ''
  type: Spec
```

Also from the code I don't see any code actually using it, seems to be an oversight: https://github.com/airbytehq/airbyte/blob/e029353b9f1728b3d1fc04471347eb033fb97075/airbyte-cdk/python/airbyte_cdk/sources/declarative/stream_slicers/substream_slicer.py#L35

The way I made it work is to use the `stream_slice` variable in jinja interpolation like you can see in the example manifest


Adjusted the PR to not list the `path` option

Reply:
Actually, it's even a bit worse than that, because it looks like for the other injection mechanisms it's ignoring the field name (always using the `stream_slice_field`), but it requires it by the schema definition: https://github.com/airbytehq/airbyte/blob/e029353b9f1728b3d1fc04471347eb033fb97075/airbyte-cdk/python/airbyte_cdk/sources/declarative/stream_slicers/substream_slicer.py#L103

@brianjlai could you take a look at that? Seems like it should be cleaned up one way or another.

Reply:
For example for such a slicer:
<img width="860" alt="Screenshot 2023-01-05 at 10 58 24" src="https://user-images.githubusercontent.com/1508364/210752884-ce6bfb81-6d91-4bc4-bf3a-d468844f8aee.png">

I would expect the name from the parent record to be injected into the `abc` header, but it's actually injected into the `the_renamed_value_to_interpolate_somewhere_else` header:
```
{
  "url": "https://pokeapi.co/api/v2/pokemon/bulbasaur",
  "parameters": null,
  "body": null,
  "headers": {
    "User-Agent": "python-requests/2.28.1",
    "Accept-Encoding": "gzip, deflate",
    "Accept": "*/*",
    "Connection": "keep-alive",
    "the_renamed_value_to_interpolate_somewhere_else": "bulbasaur"
  },
  "http_method": "GET"
}
```

Reply:
Agreed that hiding this in the UI makes sense for now

Reply:
@flash1293 @lmossman 
> On the substream slicer you can set the request_option to inject into path, but it's not actually doing it (it's just ignored)
This is more or less a current limitation of the framework. Stream slicers don't have a concept of path since that is more of a requester concept. So if you want to hide it from the frontend that sounds fine to me because it's not being used.

> Also, you can choose to inject into request_params, header and so on but that it is requiring the field name which is then ignored (it is using the stream_slice_field instead)
As you have pointed out in everything above, this to me looks to be a bug, we should be extracting the parent value from the `stream_slice_field` and storing it into params using the `parent_config.request_option.field_name`. I think the main reason we never caught this until now is that no customers are using substream request options except Braze. And for that their `stream_slice_field` and `field_name` had the same values.

I'm going to file a ticket to cover both these cases. For the first case, throwing an error when we see any parent_stream_configs that have path specified. And for the second case, we just need to change the mapping to use the `field_name`. Thanks for highlighting this.

Reply:
@brianjlai just curious, did we capture this in an issue? Want to make sure it doesn't go untracked

Reply:
yep, I filed the ticket here: https://github.com/airbytehq/airbyte/issues/21095